core/chains/evm/txmgr/orm.go

```diff
@@ -44,17 +46,22 @@ type ORM interface {
 	InsertEthReceipt(receipt *EthReceipt) error
 	InsertEthTx(etx *EthTx) error
 	InsertEthTxAttempt(attempt *EthTxAttempt) error
+	LoadEthTxAttempts(etx *EthTx, qopts ...pg.QOpt) error
+	LoadEthTxesAttempts(etxs []*EthTx, qopts ...pg.QOpt) error
 	MarkAllConfirmedMissingReceipt(chainID big.Int) (err error)
+	MarkOldTxesMissingReceiptAsErrored(blockNum int64, finalityDepth uint32, chainID big.Int, qopts ...pg.QOpt) error
```

Comment:
There will need to be some effort to get a consistent context passing throughout all of these methods.

Some queries can be exited early when a subservice receives a close signal - for others, it is best to let them complete. This needs to be handled on a case-by-case basis.

Reply:
I think this can be easily addressed by having the ORM method take in a new parent context that is independent of the subservice. So this use case is supported by standardizing qopts.

Reply:
Is there a ticket to track doing an audit of every single query to determine which ones ought to be cancellable on component close, and which should always try to complete? If not, we should create one. Its tedious but I can help you go through these.